src/lucky_migrator/create_table_statement.cr

```diff
@@ -41,6 +57,13 @@ class LuckyMigrator::CreateTableStatement
     end
   end
 
+  # Generates raw sql for adding an index to a table column. Accepts 'unique' and 'using' options.
+  def add_index(column : String | Symbol, unique = false, using : String | Symbol = "btree")
+    raise "index type '#{using}' not supported" unless ALLOWED_INDEX_TYPES.includes?(using.to_s)
+
+    IndexDefinition.new(@table_name, column, using, unique).add_to(indices)
```

Comment:
Oh I think I misunderstood the `add_to(indices)`. I think instead it might make more sense to make it more generic. Something like this: `indices << IndexDefinition.new(...).build`. That way the IndexDefinition is a bit more flexible

Reply:
Great, I was wondering about the proper way to do it, and this way is much cleaner!